packages/app-runtime/test/runtime/Onboarding.test.ts

```diff
@@ -82,4 +84,24 @@ describe("Onboarding", function () {
         });
         expect(recoveryKitResponse2).toBeSuccessful();
     });
+
+    test("should onboard with an anonomously created empty token", async () => {
+        const emptyToken = await onboardingRuntime.anonymousServices.tokens.createEmptyToken();
+
+        const device = await services.transportServices.devices.createDevice({});
+        const onboardingInfo = await services.transportServices.devices.getDeviceOnboardingInfo({ id: device.value.id });
+
+        const tokenContent = TokenContentDeviceSharedSecret.from({ sharedSecret: DeviceMapper.toDeviceSharedSecret(onboardingInfo.value) });
+        const result = await services.transportServices.tokens.updateTokenContent({ reference: emptyToken.value.reference.truncated, content: tokenContent.toJSON() });
+        expect(result).toBeSuccessful();
+
+        const token = await onboardingRuntime.anonymousServices.tokens.loadPeerToken({ reference: emptyToken.value.reference.truncated });
+        expect(token.value.content["@type"]).toBe("TokenContentDeviceSharedSecret");
+
+        const content = token.value.content as ITokenContentDeviceSharedSecret;
+        const deviceOnboardingDTO = DeviceMapper.toDeviceOnboardingInfoDTO(TokenContentDeviceSharedSecret.from(content).sharedSecret);
+
+        const onboardResult = await onboardingRuntime.accountServices.onboardAccount(deviceOnboardingDTO);
```

Comment:
Same here: a single method that completes the onboarding process by loading the updated token and creating the device would be preferable.

Reply:
As discussed, we will do this in the app